web/webkit/src/main/scala/net/liftweb/http/Templates.scala

```diff
@@ -195,23 +183,21 @@ object Templates {
           case _ =>
             val pls = places.mkString("/", "/", "")
 
-            val se = suffixes.iterator
+            val se = parsers.iterator
             val sl = List("_" + locale.toString, "_" + locale.getLanguage, "")
 
             var found = false
             var ret: NodeSeq = null
 
             while (!found && se.hasNext) {
-              val s = se.next
+              val (suffix, parser) = se.next
               val le = sl.iterator
               while (!found && le.hasNext) {
                 val p = le.next
-                val name = pls + p + (if (s.length > 0) "." + s else "")
+                val name = pls + p + (if (suffix.length > 0) "." + suffix else "")
                 import scala.xml.dtd.ValidationException
                 val xmlb = try {
-                  LiftRules.doWithResource(name) { is =>
-                    if (s == "md") {parseMarkdown(is, needAutoSurround)} else
-                    parserFunction(is) } match {
+                  LiftRules.doWithResource(name) { parser.parse } match {
```

Comment:
Can this just be `doWithResource(name)(parser.parse)`? That seems slightly clearer to me.